### PR TITLE
Books api

### PR DESCRIPTION
**Pull request by HarryPerez**
#### Finished the integration with the back end with the next features:
* Dashboard now retrieves all the books from the server
* Book detail now retrieves all the information for the selected book

## Trello Card
https://trello.com/c/43W2X7hV/31-books-api

## Screenshots
peek 2018-02-27 16-47
peek 2018-02-27 16-46

### CODE REVIEW

File: React/wbooks/src/app/screens/BookDetail/index.js [hunk 1]
```diff
@@ -15,15 +15,10 @@ class BookDetail extends Component {
...
-  componentWillMount() {
-    const book = booksJson.filter(book => book.id == this.props.match.params.bookId );
-    if(book.length === 1){
-      this.setState({ book: book[0] });
-    }else if(book.length > 1){
-      this.setState({ error: 'Sorry, it was found more than one book with the given id.' });
-    }else {
-      this.setState({ error: 'Sorry, the book was not found.' });
-    }
+  componentWillMount = async () => {
```
Comment (wfolini) [inline React/wbooks/src/app/screens/BookDetail/index.js]: if you use `.then` and/or `.catch` and no other action `async await` aren't necessary

File: React/wbooks/src/app/screens/Home/index.js [hunk 2]
```diff
@@ -40,7 +40,11 @@ class Home extends Component {
           return book;
         }
       });
-      this.setState({ books: filteredBooks });
+      this.setState({ filteredBooks: filteredBooks });
+  }
+
+  componentWillMount = async () => {
+    await bookService.getAllBooks().then((response) => {this.setState({ 'books':response.data, 'filteredBooks': response.data })});
```
Comment (wfolini) [inline React/wbooks/src/app/screens/Home/index.js]: Same here

File: React/wbooks/src/app/screens/Login/components/LoginForm/index.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ class LoginForm extends Component {
   }
 
   validateUser = async () => {
-    await authService.retrieveUserData(this.state.name, this.state.password).then(() => {this.setState({ isLogged : true })})
+    await authService.retrieveUserData(this.state.name, this.state.password).then(() => {this.setState({ hasErrors : '' })})
```
Comment (wfolini) [inline React/wbooks/src/app/screens/Login/components/LoginForm/index.js]: Same here

File: React/wbooks/src/services/bookService.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+export const getAllBooks = async () => {
+  return await api.get('/books', {
+      'headers': { 'Authorization': retrieveUserFromSession() }
+  }).then(response => {
+    return response;
+  }).catch(error => {
+    throw error;
+  });
+}
```
Comment (WFolini) [inline React/wbooks/src/services/bookService.js]: you can simplify it in the following way...

```javascript
export const getAllBooks = () => api.get('/books');
```

then, in the screen...

```javascript
fetchBooks = async () => {
  const response = await bookServices.getAllBooks();
  if (response.ok) {
    // Do something if request is sucessfull
  } 
  else {
    // Do if error
  }
}
```

Comment (wfolini) [inline React/wbooks/src/services/bookService.js]: Other comment, set a header in each endpoint is wrong. You must set the header when login is satisfactory.

File: React/wbooks/src/services/bookService.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+export const getBookDetail = async (bookId) => {
```
Comment (wfolini) [inline React/wbooks/src/services/bookService.js]: Same comments here

**Review by wfolini — CHANGES_REQUESTED** [review]

File: React/wbooks/src/services/authService.js [hunk 6]
```diff
@@ -3,23 +3,24 @@ import api from '../app/config/api';
...
+  }
 }
 
 export const retrieveUserFromSession = () => {
   return localstorageService.retrieveUserFromLocalStorage();
 }
 
 export const registerUser = async(name, password, confirmPassword, firstName, lastName) => {
-  await api.post('/users', {
+  return await api.post('/users', {
```
Comment (wfolini) [inline React/wbooks/src/services/authService.js]: Here isn't needed the `async await`

File: React/wbooks/src/services/bookService.js [hunk 7]
```diff
@@ -7,21 +7,9 @@ export const retrieveUserFromSession = () => {
...
 
 export const getAllBooks = async () => {
-  return await api.get('/books', {
-      'headers': { 'Authorization': retrieveUserFromSession() }
-  }).then(response => {
-    return response;
-  }).catch(error => {
-    throw error;
-  });
+  return await api.get('/books');
```
Comment (wfolini) [inline React/wbooks/src/services/bookService.js]: Same here

File: React/wbooks/src/services/bookService.js [hunk 8]
```diff
@@ -7,21 +7,9 @@ export const retrieveUserFromSession = () => {
...
 
 export const getBookDetail = async (bookId) => {
-  return await api.get('/books/'+bookId, {
-      'headers': { 'Authorization': retrieveUserFromSession() }
-  }).then(response => {
-    return response;
-  }).catch(error => {
-    throw error;
-  });
+  return await api.get('/books/'+bookId);
```
Comment (wfolini) [inline React/wbooks/src/services/bookService.js]: Same

File: React/wbooks/src/app/config/api.js [hunk 9]
```diff
@@ -1,7 +1,9 @@
 import axios from 'axios';
+import * as localstorageService from '../../services/localstorageService';
 
 const api = axios.create({
-  baseURL: 'https://wbooks-api-stage.herokuapp.com/api/v1'
+  baseURL: 'https://wbooks-api-stage.herokuapp.com/api/v1',
+  headers: { 'Authorization': localstorageService.retrieveUserFromLocalStorage() }
```
Comment (wfolini) [inline React/wbooks/src/app/config/api.js]: Are u setting all user object to the header?? that's not right! just the token is needed!

File: React/wbooks/src/services/bookService.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+export const retrieveUserFromSession = () => {
+  return localStorage.retrieveUserFromLocalStorage();
+}
+
+export const getAllBooks = async () => {
+  return await api.get('/books');
+}
+
+export const getBookDetail = async (bookId) => {
+  return await api.get('/books/'+bookId);
```
Comment (wfolini) [inline React/wbooks/src/services/bookService.js]: same here

File: React/wbooks/src/services/bookService.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import api from '../app/config/api';
+
+import * as localStorage from './localstorageService.js';
+
+export const retrieveUserFromSession = () => {
+  return localStorage.retrieveUserFromLocalStorage();
+}
+
+export const getAllBooks = async () => {
+  return await api.get('/books');
```
Comment (wfolini) [inline React/wbooks/src/services/bookService.js]: same here

**Review by wfolini — CHANGES_REQUESTED** [review]

**Review by wfolini — APPROVED** [review]

**Pull request merged**